File: 171_excel_sheet_column_number/try.cpp
class Solution {
  /*
   * Complexity: O(n)
   * Space: O(1)
   */
public:
    int titleToNumber(string s) {
        int n = s.length();
        int ans = 0;
        int digit = 1;
        for(int i = n-1; i >= 0; --i){
            ans += (s[i]-'A'+ 1) * digit;
            digit *= 26;
        }
        return ans;
    }
};

File: 482_license_key_formatting/try.py
class Solution:
    """
    Complexity: O(n)
    Space: O(n)
    """
    def licenseKeyFormatting(self, S, K):
        """
        :type S: str
        :type K: int
        :rtype: str
        """
        group_list = S.upper().split('-')
        total_count = sum([len(g) for g in group_list])


        strings = ''.join(group_list)
        #move the number equal to the remainder to the first group
        new_string = strings[:total_count % K]
        if(len(new_string)):
            new_string += '-'
        # get each groups of string     
        for i in range(total_count % K, len(strings), K):        
            new_string += strings[i:i+K]
            new_string += '-'
        new_string = new_string[:-1]
            
        return new_string
